File: 101-250/128-longest-consecutive-sequence.cpp
int longestConsecutive(vector<int>& nums) {
    unordered_set<int> hash;
    for (const auto &num : nums) {
        hash.insert(num);
    }
    int ans = 0;
    while (!hash.empty()) {
        int curr = *(hash.begin());
        hash.erase(curr);
        int next = curr+1, prev = curr-1;
        while (hash.count(next)) {
            hash.erase(next++);
        }
        while (hash.count(prev)) {
            hash.erase(prev--);
        }
        ans = max(ans, next - prev - 1);
    }
    return ans;
}

File: python version/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 0 or n == 1:
            return n
        nums = set(nums)
        max_len = 1
        for num in nums:
            if num-1 not in nums:
                start = num
                while num+1 in nums:
                    num += 1
                max_len = max(num-start+1, max_len)
        return max_len